gunpowder/nodes/resample.py

```diff
@@ -65,10 +65,11 @@ def setup(self):
         source_voxel_size = self.spec[self.source].voxel_size
         spec.voxel_size = self.target_voxel_size
         self.pad = Coordinate((0,)*(len(source_voxel_size) - self.ndim) + source_voxel_size[-self.ndim:])
-        spec.roi = spec.roi.grow(-self.pad, -self.pad) # Pad w/ 1 voxel per side for interpolation to avoid edge effects
-        spec.roi = spec.roi.snap_to_grid(
-            np.lcm(source_voxel_size, self.target_voxel_size),
-            mode='shrink')
+        if spec.roi.get_shape()[0]:
```

Comment:
Checking the first dimension only seems fragile. We have a nice helper function on the `Roi` class `unbounded` that will check if any dimension is unbounded.
I think the better solution would be to fix the errors that make this break on unbounded `Roi`s. I would expect an unbounded `Roi` to be able to grow/shrink by an arbitrary amount and still be unbounded (just tested and this seems to be the way it currently works). I would also expect an unbounded `Roi` to fit any voxel size, so `snap_to_grid` should also return an unbounded `Roi`. This seems to break because the numpy rounding math seems to convert to `nan`.

Reply:
Thanks, and my apologies for the delay! I've implemented using the ```Roi.unbounded()``` function to check boundedness for now. Should I instead fix the bug in ```snap_to_grid``` in the ```Roi``` code?

Reply:
Fixed the bug in ```Roi``` and removed the check in ```Resample``` in commit [4b02a36](https://github.com/funkey/gunpowder/pull/170/commits/4b02a36a79795bdf16b15e3fc2cbb5af44d1d192)